ui/src/pages/Upload.tsx

```diff
@@ -0,0 +1,212 @@
+import {
+  Button,
+  FormControl,
+  Grid,
+  InputLabel,
+  MenuItem,
+  Select,
+  TextField,
+} from "@material-ui/core/";
+import { DropzoneArea } from "material-ui-dropzone";
+import * as React from "react";
+import styled from "styled-components";
+
+import { device } from "../styles/device";
+// https://yuvaleros.github.io/material-ui-dropzone/
+
+const StyledGrid = styled(Grid)`
+  margin-top: 48px;
+  @media ${device.laptop} {
+    justify: left;
+    width: 100vw;
+  }
+`;
+
+export const Upload: React.FC = () => {
+  const [state, setState] = React.useState<{
```

Comment:
What do you think about using `useReducer` instead of `useState` here? https://reactjs.org/docs/hooks-reference.html#usereducer

Reply:
I'm thinking specifically when we have to account for error states (like if a user puts in something into a field thats wrong and we wanna indicate that there is a correct option)

Reply:
Ok yeah looking at it a bit more I think it'll be a bit cleaner to useReducer for error states instead of going one by one with the MaterialUI component, also might be useful if we recycle this page.